File: DotNet/SETPractice/DesignPattern/SingletonPattern/PrintMessage.cs
namespace DesignPattern.SingletonPattern
{
    public sealed class PrintMessage : IPrintMessage
    {
        private static readonly Lazy<PrintMessage> printMessage = new Lazy<PrintMessage>(() => new PrintMessage());
        public static PrintMessage print
        {
            get 
            { 
                return printMessage.Value; 
            }
        }

        private PrintMessage() 
        {
            Console.WriteLine("Printing Message");
        }
        
        public void Print(string message)
        {
            Console.WriteLine(message);
        }
    }
}

File: Java/SETPractice/src/main/java/DesignPattern/SingletonPattern/PrintMessage.java
package DesignPattern.SingletonPattern;

public class PrintMessage implements IPrintMessage{
    private static PrintMessage printMessage;

    private PrintMessage() {
        System.out.println("Printing Message");
    }

    public static PrintMessage GetPrintMessage() {
        if(printMessage == null) {
            synchronized (PrintMessage.class) {
                if(printMessage == null) {
                    printMessage = new PrintMessage();
                }
            }
        }
        return printMessage;
    }

    @Override
    public void Print(String message) {
        System.out.println(message);
    }
}
